function.annouce.validate.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+function validate_ipv6() {
+	// If we get an IPv6 parameter, use that, else, use client's address
+	// The IPv6 address can either be in the form of:
+	// dead:beef::1234 - i.e., raw address, in which case we use port=
+	// -or-
+	// [dead:beef::1234]:12345 - i.e., enbedded port. Unfortunately
+	// PHP has no functions for handling IPv6 addresses so we'll
+	// have to roll our own
+
+	// Check the easy case first ..
+	if ( filter_var($_GET['ipv6'], FILTER_VALIDATE_IP, FILTER_FLAG_IPV6) ) {
```

Comment:
Does this return true for separate ports with brackets, like my earlier example `ipv6 = [dead:beef::1234] port = 12345`